src/docdb/nodes.ts

```diff
@@ -91,6 +93,10 @@ export class DocDBCollectionNode implements INode {
 		return this.db.getDbLink() + '/colls/' + this.id;
 	}
 
+	clearCache(): void {
+		this.children = null;
```

Comment:
I think your logic would be simpler if this set children to an empty array `[]` instead of null